### PR TITLE
Migrated errors to be Error Utility Tool compatible

### PR DESCRIPTION
**Pull request by sayantan1413**
**Description**

This PR:

migrates the current errors to be Error Utility Tool compatible .
- Updated the Release URL
- Updated the SMI test handler
- Update the github workflow

This PR fixes #

**Notes for Reviewers**

I was trying to push in the previous PR #25  but a new PR got generated.  I am extremely sorry for the inconvenience. This is a continued work for the #25 PR

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by sayantan1413** [comment]
Hey @mgfeller please have a look at the [lint fix](https://github.com/meshery/meshery-nginx-sm/pull/27/commits/07b91ae8ebb4bc8a3c4973be58ebdb698228efb9) and [lint fix 2](https://github.com/meshery/meshery-nginx-sm/pull/27/commits/acc391efb4a88a877d060c74ab8a117cd9e5f37f). I made this changes to pass the checks. So will you please reverify once if these changes are going to breakdown any behaviour or the adapter. I have taken reference from here
- [lint fix](https://github.com/meshery/meshery-nginx-sm/pull/27/commits/07b91ae8ebb4bc8a3c4973be58ebdb698228efb9) -> https://stackoverflow.com/questions/40429072/golang-file-transfer-stops-without-any-error/40429443#40429443
- [lint fix 2](https://github.com/meshery/meshery-nginx-sm/pull/27/commits/acc391efb4a88a877d060c74ab8a117cd9e5f37f) -> https://github.com/pulumi/pulumi/pull/3446/commits/3c2b2d62fbb024116af030d6f550a1dc05a3583b

**Comment by leecalcote** [comment]
@Shreyas220 will you have a peek here?

**Review by Shreyas220 — APPROVED** [review]
looks good to me :smile:

File: internal/config/config.go [hunk 1]
```diff
@@ -140,7 +140,7 @@ func downloadBinary(platform string) (*http.Response, error) {
 	}
 
 	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("Bad status: %s", resp.Status)
+		return nil, fmt.Errorf("bad status: %s", resp.Status)
```
Comment (DelusionalOptimist) [inline internal/config/config.go:143]: I believe this can be made a meshkit error as well?

File: internal/config/config.go [hunk 2]
```diff
@@ -161,7 +161,7 @@ func installBinary(location, platform string, res *http.Response) error {
 			return err
 		}
 
-		_, err = io.Copy(out, r)
+		_, err = io.CopyN(out, r, 1024)
```
Comment (DelusionalOptimist) [inline internal/config/config.go:164]: @sayantan1413 @Shreyas220 do we have a solution other than this? This here would not copy all the bytes of the downloaded binary thus resulting into a useless one.

**Review by DelusionalOptimist — CHANGES_REQUESTED** [review]

File: internal/config/config.go (hunk 2 above)
Comment (sayantan1413) [reply internal/config/config.go:164]: Hey @DelusionalOptimist I searched a lot but could not found any. So should I revert the change here ?

Comment (DelusionalOptimist) [reply internal/config/config.go:164]: no worries @sayantan1413. Feel free to revert this to how it was eariler and put a nosec here as well.
We'll take a look into this once this adapter gets bit of a revival. ( and then we'll move away from this CLI based way anyway ).

Comment (sayantan1413) [reply internal/config/config.go:164]: Sure @DelusionalOptimist

**Comment by sayantan1413** [comment]
Hey @DelusionalOptimist the lint check fail if I change it to `copy` and also providing the nosec also.

File: internal/config/config.go [hunk 3]
```diff
@@ -160,7 +159,7 @@ func installBinary(location, platform string, res *http.Response) error {
 		if err != nil {
 			return err
 		}
-
+		// #nosec G204
```
Comment (DelusionalOptimist) [inline internal/config/config.go:162]:
```suggestion
		// Temporary. We'll get rid of this later. 
		// #nosec
```

**Review by DelusionalOptimist — APPROVED** [review]
Thanks @sayantan1413 :heavy_check_mark:

**Pull request merged**